dev/devicelab/lib/framework/adb.dart

```diff
@@ -703,3 +791,112 @@ String get adbPath {
 
   return path.absolute(adbPath);
 }
+
+class MockDevice extends Device {
```

Comment:
I would name this and the enum to `FakeDevice`/`fake`. in Dart it is generally the case that classes with `Mock` in their name extend `package:mockito`'s `Mock` class

Reply:
Just realize but should I also replace the logging strings to "fake"?